src/css-grid/lib/grid-layout.js

```diff
@@ -1733,8 +1752,10 @@ module.exports = (function(window, document) { "use strict";
 			///////////////////////////////////////////////////////////
 			var LIMIT_IS_INFINITE = 1;		
 			var infinity = 9999999.0;
-			var fullWidth = this.element.offsetWidth;
-			var fullHeight = this.element.offsetHeight;
+			var rowCount = this.growY ? this.rcMatrix.length : this.rcMatrix[0].length;
+			var colCount = this.growY ? this.rcMatrix[0].length : this.rcMatrix.length;
+			var fullWidth = this.element.offsetWidth - (this.getRowCount() - 1) * this.rowGap;
```

Comment:
I am also not sure this would not break percentage resolution, as I don't think percentage-sized tracks should care about the grid gap taking some space away, and with this change they would:
![image](https://user-images.githubusercontent.com/364405/30126401-b620f756-92f0-11e7-87fe-8808345c15a7.png)

I think the right fix is to keep fullWidth unchanged, and have fullDistributableWidth be the fixed formula above, and pass both fullSize and fullDistributableSize to computeTrackBreadth (and use fullDistributableSize where we used fullSize to mean "the full size that should be distributed")

Reply:
^ Still true and can be seen in the test case you added